modules/grafana-service/main.tf

```diff
@@ -1,181 +1,150 @@
 # ------------------------------------------------------------------------------
 # Resources
 # ------------------------------------------------------------------------------
-
 data "aws_region" "current" {}
 
 data "aws_caller_identity" "current" {}
 
-provider "null" {
-  version = "1.0"
-}
-
-provider "random" {
-  version = "2.0"
-}
-
 # ----------------------------------------
 # RDS Cluster
 # ----------------------------------------
+data "aws_ssm_parameter" "grafana_rds_username" {
+  name = "/${var.name_prefix}/rds-username"
+}
+
+data "aws_ssm_parameter" "grafana_rds_password" {
+  name = "/${var.name_prefix}/rds-password"
+}
 
 module "grafana_rds" {
-  source              = "telia-oss/rds-instance/aws"
-  version             = "0.6.1"
-  name_prefix         = "${var.name_prefix}"
-  username            = "${data.aws_ssm_parameter.grafana_rds_username.value}"
-  password            = "${data.aws_ssm_parameter.grafana_rds_password.value}"
+  source  = "telia-oss/rds-instance/aws"
+  version = "3.0.0"
+
+  name_prefix         = var.name_prefix
+  username            = data.aws_ssm_parameter.grafana_rds_username.value
+  password            = data.aws_ssm_parameter.grafana_rds_password.value
   database_name       = "grafana"
-  subnet_ids          = "${var.private_subnet_ids}"
-  vpc_id              = "${var.vpc_id}"
+  subnet_ids          = var.private_subnet_ids
+  vpc_id              = var.vpc_id
   port                = 5432
   engine              = "postgres"
-  instance_type       = "${var.rds_instance_type}"
-  allocated_storage   = "${var.rds_instance_storage}"
+  instance_type       = var.rds_instance_type
+  allocated_storage   = var.rds_instance_storage
   multi_az            = false
-  tags                = "${var.tags}"
-  skip_final_snapshot = "false"
-  snapshot_identifier = "${var.snapshot_identifier}"
+  skip_final_snapshot = false
+  snapshot_identifier = var.snapshot_identifier
+  tags                = var.tags
 }
 
 # ----------------------------------------
 # Grafana ECS Fargate Service
 # ----------------------------------------
+# Note: The provider cannot be removed without having a "migration release" since it was included in the first place.
+# Attempts to do so will produce the following error: "Error: Provider configuration not present"
+provider "null" {}
```

Comment:
This has to be a part of the `v2.0.0` release, but the plan is to remove it and release `v2.0.1` immediately afterwards. So to upgrade you first go `v2` and then `>= v2.0.1`.